File: backend/app/schemas/ai.py
from typing import Literal

from pydantic import BaseModel

RewriteMode = Literal["expand", "shorten", "examples", "format"]


class AiTextRequest(BaseModel):
    text: str
    workspace_id: str | None = None
    note_id: str | None = None


class AiRewriteRequest(BaseModel):
    text: str
    mode: RewriteMode
    workspace_id: str | None = None
    note_id: str | None = None

File: backend/src/types/ai.ts
export type RewriteMode = 'expand' | 'shorten' | 'examples' | 'format';

export interface AiTextRequest {
  text: string;
  workspace_id?: string | null;
  note_id?: string | null;
}

export interface AiRewriteRequest {
  text: string;
  mode: RewriteMode;
  workspace_id?: string | null;
  note_id?: string | null;
}
